### PR TITLE
Extend acquireTokenSilent Instrumentation

### PR DESCRIPTION
**Pull request by jo-arroyo**
This PR adds instrumentation through logger messages to acquireTokenSilent for msal-core.

### CODE REVIEW

File: lib/msal-core/src/UserAgentApplication.ts [hunk 1]
```diff
@@ -757,12 +779,15 @@ export class UserAgentApplication {
             }
         })
             .then(res => {
+                this.logger.verbose("Successfully acquired token");
                 this.telemetryManager.stopAndFlushApiEvent(request.correlationId, apiEvent, true);
+                this.logger.info("Stopping telemetryManager and flushing API event");
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:784]: I'm working in the telemetry manager now, and will add these sorts of logs there, so you can skip these here.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 2]
```diff
@@ -708,27 +724,33 @@ export class UserAgentApplication {
...
 
-                // start http event
+                // start http event TODO: add why
                 return serverAuthenticationRequest.authorityInstance.resolveEndpointsAsync(this.telemetryManager, request.correlationId)
                     .then(() => {
                         /*
                          * refresh attempt with iframe
                          * Already renewing for this scope, callback when we get the token.
                          */
+                        this.logger.verbose(`Updated authority: ${JSON.stringify(serverAuthenticationRequest.authorityInstance)}`);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:751]: How big is this object? A more succinct might be better.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 3]
```diff
@@ -708,27 +724,33 @@ export class UserAgentApplication {
...
                 if (!serverAuthenticationRequest.authorityInstance) {
                     serverAuthenticationRequest.authorityInstance = request.authority ? AuthorityFactory.CreateInstance(request.authority, this.config.auth.validateAuthority) : this.authorityInstance;
+                    if (!request.authority){
+                        this.logger.verbose("Using default authority since no authority was passed to API");
+                    }
                 }
                 // cache miss
 
-                // start http event
+                // start http event TODO: add why
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:744]: I'm working on this and can explain provide the context.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 4]
```diff
@@ -668,23 +678,29 @@ export class UserAgentApplication {
                 request.correlationId,
             );
 
+            this.logger.verbose(`ServerAuthenticationRequest: ${JSON.stringify(serverAuthenticationRequest)}`);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:681]: I think providing succinct, specific messages will be more helpful than stringify entire objects. What sort of specific pieces of information might helpful to know here?

File: lib/msal-core/src/UserAgentApplication.ts [hunk 5]
```diff
@@ -630,32 +630,42 @@ export class UserAgentApplication {
...
 
             const scope = request.scopes.join(" ").toLowerCase();
+            this.logger.verbose(`Serialized scopes: ${scope}`);
 
             // if the developer passes an account, give that account the priority
             const account: Account = request.account || this.getAccount();
+            this.logger.verbosePii(`Set account to: ${JSON.stringify(account)}`);
 
-            // extract if there is an adalIdToken stashed in the cache
+            // extract if there is an adalIdToken stashed in the cache TODO: Ask about why
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:652]: This allows MSAL to SSO to an existing session that was started with ADAL and upgrade it to an MSAL session. This allows for seemless SSO when upgrading an app from ADAL -> MSAL.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 6]
```diff
@@ -630,32 +630,42 @@ export class UserAgentApplication {
...
         const requestSignature = RequestUtils.createRequestSignature(request);
 
         return new Promise<AuthResponse>((resolve, reject) => {
 
-            // block the request if made from the hidden iframe
+            // block the request if made from the hidden iframe TODO: Ask about why to extend comment
             WindowUtils.blockReloadInHiddenIframes();
 
             const scope = request.scopes.join(" ").toLowerCase();
+            this.logger.verbose(`Serialized scopes: ${scope}`);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:646]: I believe this should be PII since it could contain client ID.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 1 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:784]: Removed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 6 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:646]: Updated.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 2 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts]: Changed logger to no longer print object but simply state that authority has been updated

File: lib/msal-core/src/UserAgentApplication.ts (hunk 4 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts]: Agreed. Changed logger to no longer print out object, but simply state that a server authentication request was built.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 5 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts]: Updated comment to reflect the above.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 3 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts]: Removed TODOs for now. Will revisit when working on code comments.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 7]
```diff
@@ -630,6 +630,8 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(userRequest: AuthenticationParameters): Promise<AuthResponse> {
+        this.logger.info("AcquireTokenSilent has been called");
+        this.logger.verbose(`AcquireTokenSilent params:\n${JSON.stringify(userRequest)}`);
```
Comment (jo-arroyo) [inline lib/msal-core/src/UserAgentApplication.ts:634]:
```suggestion
```

File: lib/msal-core/src/UserAgentApplication.ts [hunk 8]
```diff
@@ -680,11 +697,14 @@ export class UserAgentApplication {
...
             }
 
+            this.logger.verbose("Query parameters populated");
+
             const userContainedClaims = request.claimsRequest || serverAuthenticationRequest.claimsValue;
 
             let authErr: AuthError;
             let cacheResultResponse;
 
+            // If request.forceRefresh is set to true, we want to force a request for a new token instead of getting it from the cache
```
Comment (jo-arroyo) [inline lib/msal-core/src/UserAgentApplication.ts:707]:
```suggestion
            // If request.forceRefresh is set to true, force a request for a new token instead of getting it from the cache
```

File: lib/msal-core/src/UserAgentApplication.ts [hunk 9]
```diff
@@ -641,21 +642,34 @@ export class UserAgentApplication {
...
+             * In the event of no account being passed in the config, no session id, and no pre-existing adalIdToken, user will need to log in
+             */
             if (!account && !(request.sid  || request.loginHint) && StringUtils.isEmpty(adalIdToken) ) {
                 this.logger.info("User login is required");
+                // The promise rejects with a UserLoginRequiredError, which should be caught and user should be prompted to log in interactively
                 return reject(ClientAuthError.createUserLoginRequiredError());
             }
 
             // set the response type based on the current cache status / scopes set
             const responseType = this.getTokenType(account, request.scopes, true);
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts:671]: May be worth adding logging inside this function as well.

Comment (jasonnutter) [reply lib/msal-core/src/UserAgentApplication.ts:671]: They are just focusing on the `acquireTokenSilent` function for this PR. They will incrementally add logs for rest of UAA in future PRs.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 10]
```diff
@@ -630,6 +630,7 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(userRequest: AuthenticationParameters): Promise<AuthResponse> {
+        this.logger.info("AcquireTokenSilent has been called");
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:633]: I think logs like this at the start of APIs should be verbose.

Comment (sameerag) [reply lib/msal-core/src/UserAgentApplication.ts:633]: Or at-least highlight what is called, `request` specifics if it helps with log analysis, not just the API name. Depending on PII settings ofcourse.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 11]
```diff
@@ -641,21 +642,34 @@ export class UserAgentApplication {
...
 
             // if the developer passes an account, give that account the priority
-            const account: Account = request.account || this.getAccount();
+            let account: Account;
+            if (request.account) {
+                account = request.account;
+                this.logger.verbose("Account set from request");
+            } else {
+                account = this.getAccount();
+                this.logger.verbose("Account set from MSAL configuration");
```
Comment (sameerag) [inline lib/msal-core/src/UserAgentApplication.ts:654]: Can we please rephrase this as `Account set from MSAL Cache`?

File: lib/msal-core/src/UserAgentApplication.ts [hunk 12]
```diff
@@ -708,17 +727,20 @@ export class UserAgentApplication {
             else {
                 let logMessage;
                 if (userContainedClaims) {
-                    logMessage = "Skipped cache lookup since claims were given.";
+                    logMessage = "Skipped cache lookup since claims were given";
                 } else if (request.forceRefresh) {
                     logMessage = "Skipped cache lookup since request.forceRefresh option was set to true";
                 } else {
-                    logMessage = "Token is not in cache for scope:" + scope;
+                    logMessage = "Token is not in cache for scope: " + scope;
```
Comment (sameerag) [inline lib/msal-core/src/UserAgentApplication.ts:734]: Why is the `if` case `verbose` and `else` case `info` for logs here?

Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:734]: @sameerag This is unclear. These logs were already here, we just changed the punctuation. Do you feel that info or verbose is more appropriate?

File: lib/msal-core/src/UserAgentApplication.ts (hunk 9 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:671]: Noted, will add logs when we get to those functions.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 13]
```diff
@@ -641,21 +642,34 @@ export class UserAgentApplication {
...
+                this.logger.verbose("Account set from MSAL configuration");
+            }
 
-            // extract if there is an adalIdToken stashed in the cache
+            // Extract adalIdToken if stashed in the cache to allow for seamless ADAL to MSAL migration
             const adalIdToken = this.cacheStorage.getItem(Constants.adalIdToken);
 
-            // if there is no account logged in and no login_hint/sid is passed in the request
+            /**
+             * if there is no account logged in and no login_hint/sid is passed in the request
```
Comment (tnorling) [inline lib/msal-core/src/UserAgentApplication.ts:661]: Can we remove this line since it's explained again below?

File: lib/msal-core/src/UserAgentApplication.ts [hunk 14]
```diff
@@ -708,17 +727,20 @@ export class UserAgentApplication {
...
+                    logMessage = "Token is not in cache for scope: " + scope;
                 }
                 this.logger.verbose(logMessage);
 
-                // Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the api.
+                // Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the API.
                 if (!serverAuthenticationRequest.authorityInstance) {
                     serverAuthenticationRequest.authorityInstance = request.authority ? AuthorityFactory.CreateInstance(request.authority, this.config.auth.validateAuthority) : this.authorityInstance;
+                    if (!request.authority){
+                        this.logger.verbose("Using default authority since no authority was passed to API");
```
Comment (tnorling) [inline lib/msal-core/src/UserAgentApplication.ts:742]: Could we instead log the authority being used regardless of default or not? Can grab it from the instance created.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 15]
```diff
@@ -745,7 +769,7 @@ export class UserAgentApplication {
                                 this.renewIdToken(requestSignature, resolve, reject, account, serverAuthenticationRequest);
                             } else {
                                 // renew access token
-                                this.logger.verbose("renewing accesstoken");
+                                this.logger.verbose("renewing access token");
```
Comment (tnorling) [inline lib/msal-core/src/UserAgentApplication.ts:772]: nit:
```suggestion
                                this.logger.verbose("Renewing access token");
```

File: lib/msal-core/src/UserAgentApplication.ts (hunk 14 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:742]: Sure. Change made.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 13 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:661]: Good catch! Removed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 15 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:772]: Thanks! Changed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 12 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:734]: Chose to change to verbose for now.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 11 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:654]: Sure! Changed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 10 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:633]: Noted. Changed in commit.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 16]
```diff
@@ -630,6 +630,8 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(userRequest: AuthenticationParameters): Promise<AuthResponse> {
+        this.logger.verbose("AcquireTokenSilent has been called");
+        this.logger.verbose(`AcquireTokenSilent params:\n${JSON.stringify(userRequest)}`);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:634]: This could include Pii. I would argue it can be removed, since relevant parts of the request are logged downstream.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 17]
```diff
@@ -708,18 +725,20 @@ export class UserAgentApplication {
...
+                    logMessage = "Token is not in cache for scope: " + scope;
                 }
                 this.logger.verbose(logMessage);
 
-                // Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the api.
+                // Cache result can return null if cache is empty. In that case, set authority to default value if no authority is passed to the API.
                 if (!serverAuthenticationRequest.authorityInstance) {
                     serverAuthenticationRequest.authorityInstance = request.authority ? AuthorityFactory.CreateInstance(request.authority, this.config.auth.validateAuthority) : this.authorityInstance;
                 }
+                this.logger.verbose(`Authority instance: ${serverAuthenticationRequest.authority}`);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:740]: This could include the client ID, so should be `verbosePii`.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 18]
```diff
@@ -668,6 +680,8 @@ export class UserAgentApplication {
                 request.correlationId,
             );
 
+            this.logger.verbose("Finished building server authentication request");
+
             // populate QueryParameters (sid/login_hint) and any other extraQueryParameters set by the developer
             if (ServerRequestParameters.isSSOParam(request) || account) {
                 serverAuthenticationRequest.populateQueryParams(account, request, null, true);
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:687]: Can we add a log statement inside this `if`?

File: lib/msal-core/src/UserAgentApplication.ts [hunk 19]
```diff
@@ -680,11 +694,14 @@ export class UserAgentApplication {
                 serverAuthenticationRequest.populateQueryParams(account, null, adalIdTokenObject, true);
             }
 
+            this.logger.verbose("Query parameters populated");
```
Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts:697]: Instead of this, I would add an `else` to the `if/else if` directly above then add a statement indicating no additional query parameters added.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 16 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:634]: Agreed. Removed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 17 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:740]: Makes sense. Changed.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 19 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:697]: Noted. Changes made.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 18 above)
Comment (jo-arroyo) [reply lib/msal-core/src/UserAgentApplication.ts:687]: Added log statement. Let me know if you want this statement changed.
